enhancements/update/overriding-blocked-edges/README.md

```diff
@@ -0,0 +1,58 @@
+This can make users uncomfortable.
+For example, a user might:
+
+1. See a new A -> B edge appear in their channel.
+2. Begin sending stage clusters over the edge to test its stability for their cluster flavor.
+3. Become confident in the update's stability for their flavor.
+4. Schedule a maintenance window to update their production cluster(s).
+5. Have the edge pulled from their channel, causing distress when the maintenance window arrives and the A -> B edge is no longer recommended.
+
+There have been many requests about rendering update graphs or pushing notifications about edge-pulls to mitigate these concerns, but I'm still not clear on how useful those would be.
+I understand the user's decision tree to flow something like:
+
+<div style="text-align:center">
+  <img src="flow.svg" width="100%" />
+</div>
+
+I've marked the risky flows dashed and orange and the conservative flows thick and blue.
+
+For me the sticking point is: who would ever say "yes" to "Are we confident Red Hat's reasoning is exhaustive?"?
+If you feel like there are bugs you missed in your local testing but that there are no bugs which Red Hat has missed, why are you even doing local testing?
+I've marked the "What bug(s) did Red Hat link?" node dashed and orange to show that I consider it a risky pathway that we do not need to enable with a convenient, structured response.
+
+We try to update the edge-blocking comments as we discover new reasons, but:
+
+1. There are probably additional edge-blocking reasons that we are not aware of at the moment.
+2. We don't look too hard at edges after we pull them from the graph.
+
+For pulling edges, we just need to find *one* sufficiently severe bug on the edge.
+For initiating an update, users need confidence that there are no likely, severe bugs affecting their clusters.
+These are two different things.
+
+Getting [targeted edge blocking](../targeted-update-edge-blocking.md) in place will allow us to decrease the amount of collateral damage where we currently have to pull an edge for all clusters to protect a known, vulnerable subset of clusters.
+And [alerting on available updates][alert-on-available-updates] will inform folks who have had an edge restored that they have been ignoring the (new) update opportunity.
+Between those two, and similar efforts, I don't think that broadcasting edge-pulled motivations is a useful activity, and I think encouraging users to actively consume our blocked-edge reasoning in order to green-light their own off-graph updates is actively harmful.
+Because do we support them if they hit some new bug?
+Our docs around this are [currently wiggly][off-graph-support], but I expect there will be a lot of pain if we offer blanket support for folks taking any blocked edge we have ever served in a fast/stable channel.
+
+Also note that the flow is always going through "Do local tests pass?".
```

Comment:
Not always. I expect users doing upgrade without any tests on QA or dev or pre-prod environments. If these environments pass then it will serve as a data point for them and they will upgrade the production. I think you should include this in the docs.

Reply:
The decision tree only applies to production clusters, because who cares if an expendable cluster blows up?  You can just throw it away and provision a replacement.  I've added wording to that effect with f21bfb3 -> c944cd1 :

```console
$ git --no-pager diff -U0 f21bfb3..c944cd1
diff --git a/enhancements/update/overriding-blocked-edges/README.md b/enhancements/update/overriding-blocked-edges/README.md
index bc3c209..dc3f996 100644
--- a/enhancements/update/overriding-blocked-edges/README.md
+++ b/enhancements/update/overriding-blocked-edges/README.md
@@ -14 +14 @@ There have been many requests about rendering update graphs or pushing notificat
-I understand the user's decision tree to flow something like:
+I understand the user's decision tree for production clusters to flow something like:
@@ -50,0 +51,2 @@ But if they stick to "yes" and waived support, it's their cluster and their deci
+Testing clusters and others that are easily replaceable can attempt risky updates without going through this decision tree, because if they have issues during the update or on the target release, the adminstrators can just tear down the impacted cluster and provision a replacement on the original release.
+This sort of testing with expendable clusters is how admistrators would address the "local test" portions of the production cluster decision tree.
```